### PR TITLE
Fix deprecated_call regression introduced in 2.8.4

### LINKED ISSUE #1238: pytest.deprecated_call breaks with TypeError on pytest 2.8.4
https://jenkins.cryptography.io/job/cryptography-pr-builder/4650/TOXENV=py27,label=10.10/console

shows a traceback of this occurring.

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #1238

### CODE REVIEW

File: _pytest/recwarn.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def warn_explicit(message, category, *args, **kwargs):
         categories.append(category)
         old_warn_explicit(message, category, *args, **kwargs)
 
-    def warn(message, category=None, **kwargs):
+    def warn(message, category=None, *args, **kwargs):
```
Comment (nicoddemus) [inline _pytest/recwarn.py]: Grrr! :angry:

**Comment by nicoddemus** [comment]
@alex could you please test my branch with `cryptography`? I tried to run its tests on an `ubuntu` box but didn't have much success... if you have a working environment on hand I would appreciate it if you could test it before we merge this.

```
pip install git+https://github.com/nicoddemus/pytest.git@fix-deprecated-call-args
```

**Comment by alex** [comment]
@nicoddemus Confirmed that this fixes the issue for me. Thanks!

**Comment by nicoddemus** [comment]
Thanks for testing it! :smile:

**Pull request merged**